### PR TITLE
ENG-1274 Added the log service button

### PR DESCRIPTION
**Pull request by upendranadh-cruxito**

### CODE REVIEW

**Comment by percee** [comment]
@upendranadh-cruxito May you target the develop branch for this PR?

File: src/components/logservice/LogServiceButton.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  onClick?: () => void,
+};
+
+/**
+ * LogServiceButton
+ * @param props
+ * @constructor
+ */
+export const LogServiceButton = (props: LogServiceButtonProps) => {
+  const [isOpen, setIsOpen] = React.useState(!!props.isOpen);
```
Comment (percee) [inline src/components/logservice/LogServiceButton.tsx]: Instead of managing modal state manually, you should use modals manager to do so - https://mantine.dev/others/modals/

File: src/components/logservice/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                                    </div>
+                                </>
+                            )}
+                        </form>
+                    </div>
+                </div>
+            </div>
+
+
+            {/* {showExitDialogue && (
```
Comment (percee) [inline src/components/logservice/Form.tsx]: We should removed commented out code before merging.

Comment (upendranadh-cruxito) [reply src/components/logservice/Form.tsx]: removed the commented code

File: src/components/logservice/LogServiceButton.tsx (hunk 1 above)
Comment (upendranadh-cruxito) [reply src/components/logservice/LogServiceButton.tsx]: modal manager not giving the pop can you please help with this

Comment (percee) [reply src/components/logservice/LogServiceButton.tsx]: Per our convo during 1-on-1, in order to use modals developers need to have the modal provider somewhere in the parent tree. An example is available in the codebase as well as in the mantine docs.

```
import { ModalsProvider } from '@mantine/modals';

function Demo() {
  return (
    <ModalsProvider labels={{ confirm: 'Submit', cancel: 'Cancel' }}>
      <App />
    </ModalsProvider>
  );
}
```

Comment (upendranadh-cruxito) [reply src/components/logservice/LogServiceButton.tsx]: resolved

File: stories/LogServiceButton.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        <LogServiceButton {...args} />
+      </CandidApp>
+    )
+  }
+};
+
+
+export const Default = {
+  ...Template,
+  // args: parksAndGreenForm,
```
Comment (percee) [inline stories/LogServiceButton.stories.tsx]: Please remove this comment (unless you need it).

File: src/pages/organization/OrganizationListPage.tsx
Comment (percee) [inline src/pages/organization/OrganizationListPage.tsx:1]: Please revert any changes to this file as its out of scope.

File: src/components/logservice/data.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+export type FormPropsData={
```
Comment (percee) [inline src/components/logservice/data.ts]: React props should be defined in the component file (i.e., LogServiceButton.tsx)

File: src/components/logservice/FormInput.tsx
Comment (percee) [inline src/components/logservice/FormInput.tsx:1]: Custom form input component is not needed here since the form is statically defined by you as a developer - in contrast to the LearningForm where it is styled by a third party.

File: src/App.tsx
Comment (percee) [inline src/App.tsx:1]: Please revert any changes to this file as it is out of scope.

File: stories/LogServiceButton.stories.tsx (hunk 3 above)
Comment (upendranadh-cruxito) [reply stories/LogServiceButton.stories.tsx]: removed the comment

File: src/components/logservice/FormInput.tsx
Comment (upendranadh-cruxito) [reply src/components/logservice/FormInput.tsx:1]: resolved

**Review by percee — APPROVED** [review]
